### PR TITLE
Add a `minTileHeight` to ListTile widget so its height can be customized to less than the default height.

### LINKED ISSUE #145369: Add a minTileHeight to ListTile widget so its height can be customized to less than the default height
### Use case

I want to customize the listTilewidget with a height = 32
image

### Proposal

/

### PR DESCRIPTION
**Pull request by hannah-hyj**
fixes: https://github.com/flutter/flutter/issues/145369

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Review by QuncCccccc — COMMENTED** [review]
Are there any issues that this PR is trying to fix? I'm just thinking whether we can make a smaller height by changing `visualDensity` and `isDense`. Also if we need to add this property, should we also add visual density adjustment to `minTileHeight` and add this property to `ListTileThemeData`?

**Comment by hangyujin** [comment]
Added a issue.

I tried isDense and the tile height is 48, i want to implement a tile with height to 32 for my use case.

Also if we need to add this property, should we also add visual density adjustment to minTileHeight? >> I think if you set the minHeight explicitly, it makesense to override the visual density.

File: packages/flutter/test/material/list_tile_test.dart [hunk 1]
```diff
@@ -1802,6 +1802,36 @@ void main() {
     expect(tester.getSize(find.byType(ListTile)), const Size(800.0, 56.0));
     expect(right('title'), 708.0);
   });
+  testWidgets('ListTile minTileHeight', (WidgetTester tester) async {
```
Comment (QuncCccccc) [inline packages/flutter/test/material/list_tile_test.dart:1805]: Can we also update list_tile_theme_test.dart?

**Review by QuncCccccc — COMMENTED** [review]
This looks great! Just left a couple of small comments then we are good to go!

File: packages/flutter/lib/src/material/list_tile_theme.dart [hunk 2]
```diff
@@ -446,6 +461,12 @@ class ListTileTheme extends InheritedTheme {
   /// [ListTileThemeData.minLeadingWidth] property instead.
   double? get minLeadingWidth => _data != null ? _data.minLeadingWidth : _minLeadingWidth;
 
+  /// Overrides the default value of [ListTile.minTileHeight].
+  ///
+  /// This property is obsolete: please use the [data]
```
[the remark this answers is not in the document]
Comment (hangyujin) [reply packages/flutter/lib/src/material/list_tile_theme.dart:466]: Actually I just copy/pasted from other properties.
looks like `data` is preferred?

[the remark this answers is not in the document]
Comment (QuncCccccc) [reply packages/flutter/lib/src/material/list_tile_theme.dart:466]: Oh yes, probably we should move this to `data`?

[the remark this answers is not in the document]
Comment (hangyujin) [reply packages/flutter/lib/src/material/list_tile_theme.dart]: done

File: packages/flutter/lib/src/material/list_tile.dart [hunk 3]
```diff
@@ -669,6 +670,17 @@ class ListTile extends StatelessWidget {
   /// that is also null, then a default value of 40 is used.
   final double? minLeadingWidth;
 
+  /// {@template flutter.material.ListTile.minTileHeight}
+  /// The minimum height allocated for the [ListTile] widget.
+  ///
+  /// If this is null, default tile heights are 56.0, 72.0, and 88.0 for one,
+  /// two, and three lines of text respectively. If 'isDensity' is true, these
```
Comment (QuncCccccc) [inline packages/flutter/lib/src/material/list_tile.dart:677]:
```suggestion
  /// two, and three lines of text respectively. If [isDensity] is true, these
```

File: packages/flutter/lib/src/material/list_tile.dart [hunk 4]
```diff
@@ -669,6 +670,17 @@ class ListTile extends StatelessWidget {
...
+  ///  defaults are changed to 48.0, 64.0, and 76.0. A visual density value or
```
Comment (QuncCccccc) [inline packages/flutter/lib/src/material/list_tile.dart:678]:
```suggestion
  /// defaults are changed to 48.0, 64.0, and 76.0. A visual density value or
```

File: packages/flutter/lib/src/material/list_tile.dart [hunk 5]
```diff
@@ -669,6 +670,17 @@ class ListTile extends StatelessWidget {
...
+  ///  a large title will also adjust the default tile heights.
```
Comment (QuncCccccc) [inline packages/flutter/lib/src/material/list_tile.dart:679]:
```suggestion
  /// a large title will also adjust the default tile heights.
```

File: packages/flutter/lib/src/material/list_tile.dart [hunk 6]
```diff
@@ -669,6 +670,17 @@ class ListTile extends StatelessWidget {
...
+  ///
```
Comment (QuncCccccc) [inline packages/flutter/lib/src/material/list_tile.dart:680]:
```suggestion
```

**Review by QuncCccccc — APPROVED** [review]
LGTM with some nits.

**Pull request merged**